parser-typechecker/src/Unison/Codebase/Branch/Type.hs

```diff
@@ -88,4 +113,199 @@ history :: Iso' (Branch m) (UnwrappedBranch m)
 history = iso _history Branch
 
 edits :: Lens' (Branch0 m) (Map NameSegment (PatchHash, m Patch))
-edits = lens _edits (\b0 e -> b0 {_edits = e})
+edits =
+  lens
+    _edits
+    ( \b0 e ->
+        b0 {_edits = e}
+          & deriveIsEmpty
+    )
+
+terms :: Lens' (Branch0 m) (Star Referent NameSegment)
+terms =
+  lens
+    _terms
+    \branch terms ->
+      branch {_terms = terms}
+        & deriveDeepTerms
+        & deriveIsEmpty
```

Comment:
Here's where we were forgetting to re-derive `isEmpty`. (see same on edits, types, and children)